### PR TITLE
feat(issue-platform): Allow generic issue platform events to be searched via issue search

### LINKED ISSUE #42039: Add new search strategy to support searching on new Profile/Generic Issues dataset

### LINKED ISSUE #42475: Issue search should be able to query the new issues dataset and return generic issues

### PR DESCRIPTION
**Pull request by wedamija**
This allows issue platform events to be searched via issues search. This is gated by the `organizations:issue-platform` flag, so that we can make sure that only sentry sees these for now.

### CODE REVIEW

File: src/sentry/issues/search.py [hunk 1]
```diff
@@ -190,14 +190,46 @@ def _query_params_for_perf(
...
+    organization = Organization.objects.filter(id=organization_id).first()
+    if organization and features.has("organizations:issue-platform", organization=organization):
+        if group_ids:
+            filters = {"group_id": sorted(group_ids), **filters}
+
+        params = query_partial(
+            dataset=snuba.Dataset.IssuePlatform,
+            selected_columns=selected_columns,
+            filter_keys=filters,
+            conditions=[],
```
Comment (rjo100) [inline src/sentry/issues/search.py:213]: do we not want to pass conditions yet?

File: tests/snuba/search/test_backend.py [hunk 2]
```diff
@@ -2318,6 +2320,157 @@ def test_compound_message_negation(self):
...
+                search_filter_query="issue.category:profile my_tag:1",
+                sort_by="date",
+                limit=1,
+                cursor=results.next,
+                count_hits=True,
+            )
+        assert list(results) == []
+        assert results.hits == 2
+
+
```
Comment (rjo100) [inline tests/snuba/search/test_backend.py:2473]: can we add a test for the `null` condition where the feature isn't enabled? just to make sure it's all working OK

**Review by rjo100 — APPROVED** [review]

File: src/sentry/issues/search.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/issues/search.py:213]: Hmm, I think we do. Looks like my test didn't catch this, I'll update it and fix this as well

Comment (wedamija) [reply src/sentry/issues/search.py:213]: Tried to fix this, but we're not writing all the fields yet: https://github.com/getsentry/snuba/blob/8a4541f5a0d2bc2e918c25cbfa132ae84ce6a5d9/snuba/datasets/processors/search_issues_processor.py#L128

I'll merge as is without the conditions, and put up a separate pr that'll fix it once we're writing them.
